packages/agent/src/orchestrate/realize/orchestrateRealize.ts

```diff
@@ -40,44 +40,95 @@ export const orchestrateRealize =
     );
 
     if (codes.length) {
-      const files = {
-        ...ctx.state().interface?.files,
-        ...codes
-          .map((code) => ({ [code.location]: code.content }))
-          .reduce((acc, cur) => Object.assign(acc, cur), {}),
-      };
+      if (codes.every((code) => code !== FAILED)) {
+        const files = {
+          ...ctx.state().interface?.files,
+          ...codes
+            .map((code) => ({ [code.location]: code.content }))
+            .reduce((acc, cur) => Object.assign(acc, cur), {}),
+        };
 
-      const compiled = await ctx.compiler.typescript.compile({ files });
+        const compiled = await ctx.compiler.typescript.compile({ files });
 
-      return {
-        id: v4(),
-        type: "realize",
-        completed_at: new Date().toISOString(),
-        created_at: new Date().toISOString(),
-        compiled,
-        files,
-        reason: "",
-        step: ctx.state().analyze?.step ?? 0,
-      };
+        const now = new Date().toISOString();
+        ctx.dispatch({
+          type: "realizeComplete",
+          compiled: compiled,
+          created_at: now,
+          files: files,
+          step: ctx.state().analyze?.step ?? 0,
+        });
+
+        return {
+          id: v4(),
+          type: "realize",
+          completed_at: now,
+          created_at: now,
+          compiled,
+          files,
+          reason: props.reason,
+          step: ctx.state().analyze?.step ?? 0,
+        } satisfies AutoBeRealizeHistory;
+      } else {
+        const total = codes.length;
+        const failedCount = codes.filter((code) => code === FAILED).length;
+        const successCount = total - failedCount;
+
+        const now = new Date().toISOString();
+        ctx.dispatch({
+          type: "assistantMessage",
+          text: [
+            `Out of ${total} code blocks, ${successCount} succeeded, but ${failedCount} failed.`,
+            `The process has been stopped due to the failure. Please review the failed steps and try again.`,
+          ].join("\n"),
+          created_at: now,
+        });
+
+        return {
+          id: v4(),
+          type: "assistantMessage",
+          completed_at: now,
+          created_at: now,
+          text: [
+            `Out of ${total} code blocks, ${successCount} succeeded, but ${failedCount} failed.`,
+            `The process has been stopped due to the failure. Please review the failed steps and try again.`,
+          ].join("\n"),
+        } satisfies AutoBeAssistantMessageHistory;
+      }
```

Comment:
Why don't we modify this part to return the Failed File? not assistantMessage.

Reply:
interesting. but if you want to do, you might want to make other pull request for it.